include/io.tcc

```diff
@@ -186,3 +190,128 @@ void IO<Tdim>::write_volumes(const std::map<unsigned, double>& volumes) {
+      }
+      stress_vtk_file << '\n';
+    }
+
+    stress_vtk_file << "CELLS " << num_points << " " << 2 * num_points << '\n';
+    for (unsigned i = 0; i < num_points; i++)
+      stress_vtk_file << "1 " << i << '\n';
+
+    stress_vtk_file << "CELL_TYPES " << num_points << '\n';
+    for (unsigned i = 0; i < num_points; i++) stress_vtk_file << "1 " << '\n';
+
+    stress_vtk_file << "POINT_DATA " << num_points << '\n';
+    stress_vtk_file << "VECTORS Stress float\n";
+    for (auto const& stress : stresses) {
+      if (Tdim == 2)
+        stress_vtk_file << stress[0] << ' ' << stress[1] << " 0.\n";
+      else if (Tdim == 3)
+        stress_vtk_file << stress[0] << ' ' << stress[1] << ' ' << stress[2]
+                        << '\n';
+    }
+
+    stress_vtk_file.close();
+  }
+  std::cout << "Wrote initial stresses vtk file\n";
+}
+
+//! \brief Output .vtk files for viewing mesh
+//! \details Write mesh coordinates and elements
+//! \tparam Tdim dimension
+template <unsigned Tdim>
+void IO<Tdim>::write_vtk_mesh(const std::vector<Eigen::VectorXd>& vertices,
+                              const std::vector<Eigen::VectorXd>& elements) {
+
+  unsigned num_vertices, num_elements;
+  num_vertices = vertices.size();
+  num_elements = elements.size();
+
+  //! Assign element_type from VTK documentation
+  //! 5 - Triangle (3 nodes)
+  //! 9 - Quadrangle (4 nodes)
+  //! 10 - Tetrahedron (4 nodes)
+  //! 12 - Hexahedron (8 nodes)
+  //! For more informtion on element types, visit:
+  //! https://www.vtk.org/wp-content/uploads/2015/04/file-formats.pdf
+  unsigned element_type = 12;
+  unsigned nodes_in_element = 8;
+
+  std::cout << "output .vtk file for mesh will be stored in: "
+            << mesh_vtk_filename_.string() << "\n";
+
+  //! Output stress file
+  std::fstream mesh_vtk_file;
+  mesh_vtk_file.open(mesh_vtk_filename_.string(), std::ios::out);
+  //! Write .vtk file
+  if (mesh_vtk_file.is_open()) {
+    mesh_vtk_file << "# vtk DataFile Version 2.0\n";
+    mesh_vtk_file << "MPM_POINT_GENERATOR: Mesh\n";
+    mesh_vtk_file << "ASCII\n";
+    mesh_vtk_file << "DATASET UNSTRUCTURED_GRID\n";
+
+    mesh_vtk_file << "POINTS " << num_vertices << " float\n";
+    for (auto const& vertice : vertices) {
+      for (unsigned i = 0; i < vertice.size(); ++i) {
+        mesh_vtk_file << vertice[i] << ' ';
+      }
+      mesh_vtk_file << '\n';
+    }
+
+    // Note minus 1 is because element starts from 1 instead of 0
+    mesh_vtk_file << "CELLS " << num_elements << " "
+                  << num_elements * (nodes_in_element + 1) << '\n';
+    for (auto const& element : elements) {
+      mesh_vtk_file << nodes_in_element << ' ';
+      for (unsigned i = 0; i < element.size(); ++i)
+        mesh_vtk_file << element[i] - 1 << ' ';
+      mesh_vtk_file << '\n';
+    }
+
+    mesh_vtk_file << "CELL_TYPES " << num_elements << '\n';
+    for (unsigned i = 0; i < num_elements; i++)
+      mesh_vtk_file << element_type << '\n';
```

Comment:
What about 2D? These 2 types aren't the same.

Reply:
yes, that's why we have line 269-275,
or we can use 2D use `9` and 3D use `12`
or we can map them from the element in `element`

Reply:
This is good, sorry, I confused GMSH cell types with VTK.